lib/db.js

```diff
@@ -0,0 +1,384 @@
+'use strict';
+
+var util = require('util');
+var chainlib = require('chainlib');
+var BaseDB = chainlib.DB;
+var Transaction = require('./transaction');
+var async = require('async');
+var bitcore = require('bitcore');
+var BufferWriter = bitcore.encoding.BufferWriter;
+var errors = require('./errors');
+var levelup = chainlib.deps.levelup;
+var log = chainlib.log;
+var PublicKey = bitcore.PublicKey;
+var Address = bitcore.Address;
+
+function DB(options) {
+  if(!options) {
+    options = {};
+  }
+
+  BaseDB.call(this, options);
+
+  this.coinbaseAddress = options.coinbaseAddress;
+  this.coinbaseAmount = options.coinbaseAmount || 50 * 1e8;
+  this.Transaction = Transaction;
+
+  this.network = bitcore.Networks.get(options.network) || bitcore.Networks.testnet;
+}
+
+util.inherits(DB, BaseDB);
+
+DB.PREFIXES = {
+  OUTPUTS: 'outs'
+};
+DB.CONCURRENCY = 10;
+
+DB.prototype.getBlock = function(hash, callback) {
+  var self = this;
+
+  // get block from bitcoind
+  this.bitcoind.getBlock(hash, function(err, blockData) {
+    if(err) {
+      return callback(err);
+    }
+    callback(null, self.Block.fromBuffer(blockData));
+  });
+};
+
+DB.prototype.putBlock = function(block, callback) {
+  // block is already stored in bitcoind, but we need to update
+  // our prevhash index still
+  this._updatePrevHashIndex(block, callback);
```

Comment:
We may be able to expose a method from the daemon, so that the prevHash index isn't need also (perhaps in another PR?).